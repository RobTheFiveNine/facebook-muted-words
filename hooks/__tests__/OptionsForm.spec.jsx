import React from 'react';
import renderer from 'react-test-renderer';
import Options from '../OptionsForm';

describe('<Options />', () => {
  it('should render correctly', () => {
    const subject = renderer.create(<Options />);
    expect(subject).toMatchSnapshot();
  });

  describe('if the wordlist source is a URL', () => {
    it('should render correctly', () => {
      localStorage.setItem('source', 'url');
      localStorage.setItem('wordlistUrl', 'http://localhost/list.txt');

      const subject = renderer.create(<Options />);
      expect(subject).toMatchSnapshot();
    });

    describe('if the save button is clicked', () => {
      it('should save the URL in local storage', () => {
        localStorage.setItem('source', 'url');

        const subject = renderer.create(<Options />);
        const input = subject.root.findByProps({ id: 'url' });
        const button = subject.root.findByProps({ testId: 'saveButton' });

        renderer.act(() => input.props.onChange({
          target: { value: 'new url' },
        }));

        renderer.act(() => button.props.onClick());
        expect(localStorage.getItem('wordlistUrl')).toEqual('new url');
      });
    });
  });

  describe('if the wordlist source is a local list', () => {
    it('should render correctly', () => {
      localStorage.setItem('source', 'local');
      localStorage.setItem('muted', '["word1", "word2", "word3"]');

      const subject = renderer.create(<Options />);
      expect(subject).toMatchSnapshot();
    });

    describe('when the save button is clicked', () => {
      it('should serialise the words into a JSON array in local storage', () => {
        localStorage.setItem('source', 'local');

        const subject = renderer.create(<Options />);
        const input = subject.root.findByProps({ id: 'muted' });
        const button = subject.root.findByProps({ testId: 'saveButton' });

        renderer.act(() => input.props.onChange({
          target: { value: 'word1\nword2\r\nword3\nword4' },
        }));

        renderer.act(() => button.props.onClick());
        expect(JSON.parse(localStorage.getItem('muted'))).toEqual([
          'word1',
          'word2',
          'word3',
          'word4',
        ]);
      });
    });
  });
});
