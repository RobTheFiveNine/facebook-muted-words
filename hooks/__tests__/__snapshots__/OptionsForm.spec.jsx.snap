// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> if the wordlist source is a URL should render correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="source"
      >
        Source
        <select
          className="form-control"
          defaultValue="url"
          id="source"
          onChange={[Function]}
        >
          <option
            value="local"
          >
            Local List
          </option>
          <option
            value="url"
          >
            URL
          </option>
        </select>
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="muted"
      >
        Wordlist (one word per line)
        <textarea
          className="form-control"
          defaultValue=""
          id="muted"
          name="muted"
          onChange={[Function]}
          rows="10"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="url"
      >
        URL of wordlist
        <input
          className="form-control"
          defaultValue="http://localhost/list.txt"
          id="url"
          name="url"
          onChange={[Function]}
          type="text"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <button
        className="btn btn-primary btn-block"
        disabled={true}
        onClick={[Function]}
        testId="saveButton"
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`<Options /> if the wordlist source is a local list should render correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="source"
      >
        Source
        <select
          className="form-control"
          defaultValue="local"
          id="source"
          onChange={[Function]}
        >
          <option
            value="local"
          >
            Local List
          </option>
          <option
            value="url"
          >
            URL
          </option>
        </select>
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="muted"
      >
        Wordlist (one word per line)
        <textarea
          className="form-control"
          defaultValue="word1
word2
word3"
          id="muted"
          name="muted"
          onChange={[Function]}
          rows="10"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="url"
      >
        URL of wordlist
        <input
          className="form-control"
          defaultValue="new url"
          id="url"
          name="url"
          onChange={[Function]}
          type="text"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <button
        className="btn btn-primary btn-block"
        disabled={true}
        onClick={[Function]}
        testId="saveButton"
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;

exports[`<Options /> should render correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="source"
      >
        Source
        <select
          className="form-control"
          defaultValue="local"
          id="source"
          onChange={[Function]}
        >
          <option
            value="local"
          >
            Local List
          </option>
          <option
            value="url"
          >
            URL
          </option>
        </select>
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="muted"
      >
        Wordlist (one word per line)
        <textarea
          className="form-control"
          defaultValue=""
          id="muted"
          name="muted"
          onChange={[Function]}
          rows="10"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="col-md-12"
    >
      <label
        className="w-100"
        htmlFor="url"
      >
        URL of wordlist
        <input
          className="form-control"
          defaultValue=""
          id="url"
          name="url"
          onChange={[Function]}
          type="text"
        />
      </label>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <button
        className="btn btn-primary btn-block"
        disabled={true}
        onClick={[Function]}
        testId="saveButton"
        type="button"
      >
        Save
      </button>
    </div>
  </div>
</div>
`;
